src/app/common/confirmation-modal/confirm-dialog.service.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { TestBed } from '@angular/core/testing';
+
+import { ConfirmDialogService } from './confirm-dialog.service';
+import { MaterialModule } from '../../material.module';
+
+describe('ConfirmDialogService', () => {
+  let service: ConfirmDialogService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [MaterialModule],
```

Comment:
You should not import material module here. Instead you should mock `MatDialog` and test if calling `openConfirmModal` is opening the modal using `open` method and if val is mapped always to boolean